documentation/development-docs/azure-powershell-developer-guide.md

```diff
@@ -0,0 +1,451 @@
+
+```
+git remote add upstream https://github.com/Azure/azure-powershell.git
+```
+
+Then, to pull changes from the **preview** branch in _Azure/azure-powershell_ into your local working branch, run the following command:
+
+```
+git pull upstream preview
+```
+
+## Building the Environment
+
+_Note_: to build the environment locally, you need `platyPS` install on your machine. Please see the [Prerequisites](#prerequisites) section for details on how to install this module.
+
+After cloning the repository to your local machine, you want to ensure that you can build the environment. To do so, launch `VS Developer Command Prompt` (which is installed with Visual Studio) and run the following command (from the root of the repository) to do a full build:
+
+```
+msbuild build.proj
+```
+
+Another way to build the project is by using the `Repo-Tasks` module found in the repository:
+
+1. In the `tools` folder of the repository, double-click `PS-VSPrompt.lnk`
+    - This opens up a terminal that acts as PowerShell and VS Developer Command Prompt
+2. Run `Import-Module .\Repo-Tasks.psd1`
+    - This imports a [module containing cmdlets and functions that help with miscellaneous environment tasks](https://github.com/Azure/azure-powershell/blob/preview/documentation/Repo-Tasks-Module.md)
+3. Run the `Start-Build` cmdlet to do a full build of the project
+    - This builds each of the individual projects in the repository
+
+
+
+### Skipping Help Generation During Build
+
+By default, we build the `dll-Help.xml` files (used to display the help content for cmdlets in PowerShell) from markdown using the `platyPS` module. Since this help generation step can take 10-15 minutes, we have added the ability to skip it as a part of the command line build process:
+
+```
+msbuild build.proj /p:SkipHelp=true
+```
+
+_Note_: when [updating the installer](#updating-the-installer), you **should not** skip the help generation step, as it removes the `dll-Help.xml` files from the wxi file.
+
+## Running Tests
+
+With the same terminal open from the previous section, run the cmdlet `Invoke-CheckinTests` to run all of the tests in the project
+
+Alternatively, you can launch `VS Developer Command Prompt` and run the following command (from the root of the repository) to run all of the tests:
+
+```
+msbuild build.proj /t:Test
+```
+
+# Before Adding a New Project
+
+## .NET SDK
+
+Before adding a new project to Azure PowerShell, you must have generated an [SDK for .NET](https://github.com/Azure/azure-sdk-for-net) using [AutoRest](https://github.com/Azure/autorest) for your service, and it must have been merged into the repository. 
+
+For more information about on-boarding a new library in the SDK for .NET repository, click [here](https://github.com/Azure/azure-sdk-for-net#to-on-board-new-libraries).
+
+## Design Review
```

Comment:
Update this section with the new guidelines.

Reply:
@maddieclayton fixed